src/include/duckdb/common/unique_ptr.hpp

```diff
@@ -35,11 +35,12 @@ class unique_ptr : public std::unique_ptr<_Tp, _Dp> {
 		return original::get();
 	}
 
-	NOWARN(
-	    -Wattributes,
-	    // This is necessary to tell clang-tidy that it reinitializes the variable after a move
-	    [[clang::reinitializes]] inline void reset(
-	        typename original::pointer ptr = typename original::pointer()) noexcept { original::reset(ptr); })
+	DISABLE_WARN(-Wattributes)
+	// This is necessary to tell clang-tidy that it reinitializes the variable after a move
+	[[clang::reinitializes]] inline void reset(typename original::pointer ptr = typename original::pointer()) noexcept {
```

Comment:
Perhaps we can enable the `[[clang::reinitializes]]` only when doing a clang-tidy build using defines, instead of trying to disable warnings/getting this to compile on every compiler?